### PR TITLE
(ILIAS 8) T&A Bugfix #0029216: Test Import Results are being skipped

### PR DESCRIPTION
**Pull request by fhelfer**
https://mantis.ilias.de/view.php?id=29216

release_8 version of PR #6335

### CODE REVIEW

**Review by kergomard — COMMENTED** [review]
Hi @fhelfer 
Thank you very much for this PR!
I've only one small question and a small change, I will make directly inline. The question:
- [ ] How can [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6344/files#diff-84dff1fb73f8439476fe28c004986be04ddf8cf211aaf021ccb1ac5bcd0f7fb6R228) and [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6344/files#diff-84dff1fb73f8439476fe28c004986be04ddf8cf211aaf021ccb1ac5bcd0f7fb6R240) happen? This for me looks like a recipe for a lot of useless entries. As far as I understand, solutions and results need a corresponding question to be valid? What am I missing?

Thank you very much and best,
@kergomard

File: Modules/Test/classes/class.ilTestPassesSelector.php [hunk 1]
```diff
@@ -261,13 +261,15 @@ private function isReportingDateReached(): bool
...
+        $matches = [];
 
         if (!preg_match($reg, $date, $matches)) {
             return false;
         }
 
-        $repTS = mktime($matches[4], $matches[5], $matches[6], $matches[2], $matches[3], $matches[1]);
+        $matches = array_map('intval', $matches);
+
+        $repTS = mktime((int) $matches[4], $matches[5], $matches[6], $matches[2], $matches[3], $matches[1]);
```
Comment (kergomard) [inline Modules/Test/classes/class.ilTestPassesSelector.php]:
```suggestion
        $repTS = mktime($matches[4], $matches[5], $matches[6], $matches[2], $matches[3], $matches[1]);
```
We don't need the int here anymore, right?

**Comment by fhelfer** [comment]
> Hi @fhelfer Thank you very much for this PR! I've only one small question and a small change, I will make directly inline. The question:
> 
> * [x]  How can [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6344/files#diff-84dff1fb73f8439476fe28c004986be04ddf8cf211aaf021ccb1ac5bcd0f7fb6R228) and [this here](https://github.com/ILIAS-eLearning/ILIAS/pull/6344/files#diff-84dff1fb73f8439476fe28c004986be04ddf8cf211aaf021ccb1ac5bcd0f7fb6R240) happen? This for me looks like a recipe for a lot of useless entries. As far as I understand, solutions and results need a corresponding question to be valid? What am I missing?
> 
> Thank you very much and best, @kergomard

Hi @kergomard,

You are right, i have removed it for now. I had Test Export Files that had a 0 as question_fi but cannot reproduce with newl tests. It seems the export has been fixed as well.

Best, @fhelfer

**Comment by kergomard** [comment]
Thank you very much  @fhelfer .

I merge. If I'm getting this right this is ILIAS 8 only. I will also keep the corresponding Issue open, as we wait for the merge of https://github.com/ILIAS-eLearning/ILIAS/pull/6335 for trunk.

Best,
@kergomard

**Pull request merged**